### PR TITLE
Initial project setup

### PR DESCRIPTION
**Pull request by joeldenning**
This sets up a project that we can publish to npm. It is both a CLI and a javascript library. We might add a Dockerfile later so it can be used that way as well.

This is just the initial project scaffolding. I imagine much of it won't be familiar to you, but would still appreciate a review on it.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  "files": [
+    "lib"
+  ],
+  "scripts": {
+    "cli": "cross-env NODE_OPTIONS=\"--loader ts-node/esm --experimental-specifier-resolution=node\" FOUNDRY_TOKEN=sample node src/cli.ts",
+    "format": "prettier --write .",
+    "check-format": "prettier --check .",
+    "lint": "eslint \"src/**/*\"",
+    "build": "rimraf lib && tsc",
+    "prepublishOnly": "pnpm run build",
```
Comment (Qirhi) [inline package.json:17]: Is it common to use camel case for script commands?

File: src/cli.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env node
```
Comment (Qirhi) [inline src/cli.ts:1]: Is Line 1 telling the shell to execute this script using node?

File: src/cli.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  );
+}
+
+yargs(hideBin(process.argv))
+  .command(
+    "deploy <microfrontendName>",
+    "deploy a microfrontend",
+    (yargs) => {
+      return yargs.positional("microfrontendName", {
+        describe: "The name of the microfrontend you wish to deploy",
```
Comment (Qirhi) [inline src/cli.ts:20]: In my understanding of yargs, the describe option is a type of help text.

**Review by Qirhi — APPROVED** [review]
Great start on scaffolding! Very clear.  Nice work.

File: package.json (hunk 1 above)
Comment (joeldenning) [reply package.json:17]: Yes. The `prepublishOnly` script is actually a special one that has a specific meaning with npm. See https://docs.npmjs.com/cli/v8/using-npm/scripts

File: src/cli.ts (hunk 2 above)
Comment (joeldenning) [reply src/cli.ts:1]: Yes, it lets the file be executed via `./cli.js` rather than having to do `node cli.js`

File: src/cli.ts (hunk 3 above)
Comment (joeldenning) [reply src/cli.ts:20]: yes that's correct

**Pull request merged**